tests/records_test.py

```diff
@@ -501,7 +503,16 @@ def test_get_updated_records_since_date(app):
 
 
 def test_get_updated_records_on_date(app):
-    assert(type(get_inspire_records_updated_on("2021-06-27")) == list)
+    test_date = '2021-06-29'
+    mock_url = 'https://inspirehep.net/api/literature?sort=mostrecent'
+    mock_url += '&size={0}&page=1&fields=control_number'.format(RECORDS_PER_PAGE)
+    mock_url += '&q=external_system_identifiers.schema%3AHEPData%20and%20du%3A{}'.format(test_date)
+    mock_json = {'hits': {'total': 2, 'hits': [{'id': '1234567'}, {'id': '890123'}]}}
+    # Use requests_mock to mock the response from inspirehep.net.
+    with requests_mock.Mocker(real_http=True) as mock:
+        mock.get(mock_url, json=mock_json, complete_qs=True)
+        ids = get_inspire_records_updated_on(test_date)
```

Comment:
Does this test fail if you modify the URL? (I can't remember exactly how requests_mock works.) If it doesn't, I think you could use `mock.assert_called_with` (or one of the related methods to `mock.calls`) to ensure that the URL you mentioned was called.

Reply:
Yes, if `mock_url` didn't match the URL inside `get_inspire_records_updated_on`, `ids` would correspond to a real INSPIRE-HEP request, and the test would fail.

Reply:
Sorry, should have read the test more carefully!